test/computational_graph.jl

```diff
@@ -65,6 +65,25 @@
     end
 end
 
+@testset "merge_prefactors" begin
+    g1 = propagator([𝑓⁺(1), 𝑓⁻(2)])
+    h1 = ComputationalGraphs.linear_combination(g1, g1, 1, 2)
+    @test h1.subgraph_factors == [1,2]
+    h2 = ComputationalGraphs.merge_prefactors(h1)
+    @test h2.subgraph_factors == [3]
+    @test length(h2.subgraphs) == 1
+    @test isequiv(h2.subgraphs[1], g1, :id)
+    g2 = g1 * 2
+    h3 = ComputationalGraphs.linear_combination(g1, g2, 1, 2)
+    h4 = ComputationalGraphs.merge_prefactors(h3)
+    @test isequiv(h3, h4, :id)
+    h5 = ComputationalGraphs.linear_combination([g1, g2, g2, g1], [3, 5, 7, 9])
+    h6 = ComputationalGraphs.merge_prefactors(h5)
+    @test length(h6.subgraphs) ==2
+    @test h6.subgraphs == [g1,g2]
+    @test h6.subgraph_factors == [12,12]
```

Comment:
This test case looks wrong. `g2.subgraph_factors = [2]`, so in principle, the result should give `h6.subgraphs == [g1]` and `h6.subgraph_factors == [36]`. Right now it will fail, because we need to use inplace_prod to sanitize the input to merge_prefactors (in this case, `h5.subgraphs[2] -> inplace_prod(h5.subgraphs[2])` and `h5.subgraphs[3] -> inplace_prod(h5.subgraphs[3])`).

Instead of writing a passing but incorrect test, which is misleading, you should write the correct test as I mentioned above, but use `@test_broken` for these two lines. It will still allow us to merge, but mark clearly some functionality that has to be fixed/implemented before everything is working.

Reply:
I guess what you intended here was probably `g2 = propagator([𝑓⁺(1), 𝑓⁻(2)], factor=2)`, not `g2 = g1 * 2`, which is "2G<sub>1</sub>" with local factor 2. Then your current test makes sense. We should include both tests here, one with local factors which do not get merged, and one with non-local subfactors, which do get merged.

Reply:
I guess what you intended here was probably `g2 = propagator([𝑓⁺(1), 𝑓⁻(2)], factor=2)`, not `g2 = g1 * 2`, which is "2G<sub>1</sub>" with local factor 2. Then your current test makes sense. We should include both tests here, one with local factors which do not get merged, and one with non-local subfactors, which do get merged.